python/tests/integration/toolbox/test_query_stats.py

```diff
@@ -64,3 +101,396 @@ def test_query_stats_clear(s3_version_store_v1, clear_query_stats):
+    #             }
+    #         }
+    #     },
+    #     "SNAPSHOT_REF": {
+    #         "storage_ops": {
+    #             "S3_HeadObject": {
+    #                 "count": 2,
+    #                 "size_bytes": 0,
+    #                 "total_time_ms": 81
+    #             },
+    #             "S3_PutObject": {
+    #                 "count": 2,
+    #                 "size_bytes": 1171,
+    #                 "total_time_ms": 47
+    #             }
+    #         }
+    #     },
+    #     "SYMBOL_LIST": {
+    #         "storage_ops": {
+    #             "S3_DeleteObjects": {
+    #                 "count": 1,
+    #                 "size_bytes": 0,
+    #                 "total_time_ms": 65
+    #             },
+    #             "S3_GetObject": {
+    #                 "count": 1,
+    #                 "size_bytes": 309,
+    #                 "total_time_ms": 24
+    #             },
+    #             "S3_ListObjectsV2": {
+    #                 "count": 3,
+    #                 "size_bytes": 0,
+    #                 "total_time_ms": 172
+    #             },
+    #             "S3_PutObject": {
+    #                 "count": 1,
+    #                 "size_bytes": 309,
+    #                 "total_time_ms": 31
+    #             }
+    #         }
+    #     },
+    #     "VERSION_REF": {
+    #         "storage_ops": {
+    #             "S3_GetObject": {
+    #                 "count": 2,
+    #                 "size_bytes": 1218,
+    #                 "total_time_ms": 53
+    #             },
+    #             "S3_ListObjectsV2": {
+    #                 "count": 1,
+    #                 "size_bytes": 0,
+    #                 "total_time_ms": 47
+    #             }
+    #         }
+    #     }
+    # }
+    assert {"SNAPSHOT", "SNAPSHOT_REF", "SYMBOL_LIST", "LOCK", "VERSION_REF"} == stats.keys()
+    
+    assert "storage_ops" in stats["SNAPSHOT"]
+    assert "S3_ListObjectsV2" in stats["SNAPSHOT"]["storage_ops"]
+    assert "total_time_ms" in stats["SNAPSHOT"]["storage_ops"]["S3_ListObjectsV2"]
+    assert "count" in stats["SNAPSHOT"]["storage_ops"]["S3_ListObjectsV2"]
+    assert stats["SNAPSHOT"]["storage_ops"]["S3_ListObjectsV2"]["count"] == 2
+    
+    assert "storage_ops" in stats["SNAPSHOT_REF"]
+    snapshot_ref_ops = stats["SNAPSHOT_REF"]["storage_ops"]
+    
+    assert "S3_PutObject" in snapshot_ref_ops
+    assert "S3_HeadObject" in snapshot_ref_ops
+    
+    assert snapshot_ref_ops["S3_PutObject"]["count"] == 2
+    assert snapshot_ref_ops["S3_HeadObject"]["count"] == 2
+
+
+def test_query_stats_read_write(s3_version_store_v1, clear_query_stats):
+    qs.enable()
+    s3_version_store_v1.write("a", 1)
+    s3_version_store_v1.write("a", 2)
+    with config_context("VersionMap.ReloadInterval", 0):
+        s3_version_store_v1.read("a")
+        s3_version_store_v1.read("a")
+    stats = qs.get_query_stats()
+#     {
+#     "SYMBOL_LIST": {
+#         "storage_ops": {
```

Comment:
The top level grouping shouldn't be key type as William described. `storage_ops` (possibly renamed to `storage_operations` or `io`) should be at the root level, rather than key types